docs/api-reference-1/wordpress-source.md

```diff
@@ -407,80 +450,108 @@ api.init({
 });
 ```
 
-#### `api.get({ endpoint, params, api?, isWpCom? })`
+#### `libraries.api.get()`
 
 Request entity from the WordPress REST API.
 
-**arguments**
+{% hint style="info" %}
 
-* `endpoint`: name of the endpoint if is a `/wp/v2` endpoint (e.g. `posts`), or the full path of other REST endpoints (e.g. `/frontity/v1/discovery`).
-* `params`: any parameter that will be included in the query params.
-* `api` (optional): overrides the value set with `api.set.`
-* `isWpCom` (optional): overrides the value set with `api.set.`
+`(options: object) => Promise`
 
-#### return
+- **Parameters**
+  - `options: object`
+    * **`endpoint`**: `string` Name of the endpoint if is a `/wp/v2` endpoint (e.g. `posts`), or the full path of other REST endpoints (e.g. `/frontity/v1/discovery`).
+    * **`params`**: `string` Any parameter that will be included in the query params.
+    * _`api`_: `string` _(optional)_ Overrides the value set with `api.set.`
+    * _`isWpCom`_: `boolean` _(optional)_ Overrides the value set with `api.set.`
+- **Return value**
+  - `Promise` Promise resolving to data requested
 
-* A promise of type `Response`
+{% endhint %}
 
 For more info, visit the [WP REST API reference](https://developer.wordpress.org/rest-api/reference).
 
-**example**
+**Example**
 
 ```javascript
 const { api } = libraries.source;
 
 // Get posts from categories 2, 3 and 4
-api.get({ endpoint: "posts", params: { _embed: true, categories: '2,3,4' } });
+api.get({ endpoint: "posts", params: { _embed: true, categories: '2,3,4' } })
+  .then(dataRequested => { /* do something with dataRequested */ })
```

Comment:
Using promises with `.then` is pretty much deprecated in favor of async/await nowadays. I'd rather show examples using `await`:

```js
const response = await api.get({
  endpoint: "posts",
  params: { _embed: true, categories: "2,3,4" },
});
```

Reply:
> Using promises with .then is pretty much deprecated in favor of async/await nowadays. I'd rather show examples using await:

I'm not sure about this.

- I don't agree with the affirmation than the use of `.then` is deprecated
- The use of `await` alone doesn't work unless you use inside of an `async` function

Using `then` in the examples makes them ready to be "copied & pasted" while still indicates the nature of a Promise

Reply:
By deprecated I mean it's not used any more thanks to the async/await functions.

We (you and me) are used to it because we had to learn it before async/await existed. But for JS newbies the `.then` prop is really confusing. If you think about it, it's quite a horrible API compared to async/await.

Reply:
I think you're right 👍